resources/repo.rb

```diff
@@ -2,95 +2,112 @@
 property :repo_type, String, default: 'maven-hosted'
 property :attributes, Hash, default: lazy { Mash.new } # Not mandatory but strongly recommended in the generic case.
 property :online, [true, false], default: true
-property :api_url, String, required: true
-property :api_user, String, required: true
-property :api_password, String, required: true
-
-def apiclient
-  @apiclient ||= ::Nexus3::Api.new(api_url, api_user, api_password)
-end
+property :api_url, String, required: true, identity: true
+property :api_user, String, required: true, identity: true
+property :api_password, String, required: true, identity: true
 
 def textify(args)
   require 'json'
   JSON.generate(args)
 end
 
-load_current_value do
-  config_as_json = apiclient.run_script('get_repo', current_resource.repo_name)
-  config = JSON.parse(config_as_json)
-  # rubocop:disable Lint/UselessAssignment
-  repo_name = config['repositoryName']
-  repo_type = config['recipeName']
-  attributes = attributes.merge(config['attributes'])
-  online = config['online']
-  # rubocop:enable Lint/UselessAssignment
+def apiclient
+  @apiclient ||= ::Nexus3::Api.new(api_url, api_user, api_password)
+end
+
+load_current_value do |desired|
+  api_url desired.api_url
+  api_user desired.api_user
+  api_password desired.api_password
+
+  begin
+    config = apiclient.run_script('get_repo', desired.repo_name)
+    # rubocop:disable Lint/UselessAssignment
```

Comment:
This one complains because I am not using the properties that I set in this "method".